mypy/build.py

```diff
@@ -2972,8 +2972,7 @@ def dumps(self) -> str:
 def dump_timing_stats(path: str, graph: Graph) -> None:
     """Dump timing stats for each file in the given graph."""
     with open(path, "w") as f:
-        for id in sorted(graph):
-            f.write(f"{id} {graph[id].time_spent_us}\n")
+        f.writelines(f"{id} {graph[id].time_spent_us}\n" for id in sorted(graph))
```

Comment:
I find the new code harder to understand. An imperative for loop makes more sense than a generator expression here. This applies to the other `.writelines` changes below too.